.github/workflows/smoke-test-ami.yml

```diff
@@ -0,0 +1,128 @@
+name: Smoke Test ami Image
+
+concurrency:
+  group: ${{ github.workflow }}-${{github.ref}}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+defaults:
+  run:
+    shell: bash
+env:
+  AWS_REGION: "us-east-1"
+  TERRAFORM_DIRECTORY: ${{github.workspace}}/terraform
+
+jobs:
+  smoketest-ami:
+    name: PG${{ matrix.pg }}-${{ matrix.type }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # required for aws credentials
+    strategy:
+      fail-fast: false
+      matrix:
+        pg: [15]
+        type: ["alpine"]
+    steps:
+      - name: Check out the source
+        uses: actions/checkout@v3
+
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2.2.0
+        with:
+          role-to-assume: arn:aws:iam::052187009926:role/Github-Warpsql-Action
```

Comment:
Please replace the placeholder demo role with the appropriate AWS credentials to be utilized for the workflow build